project/src/main/webapp/gmail.js

```diff
@@ -15,191 +15,235 @@
+    return query;
+}
+
+class GmailAPI {
+    question;
+    emailObject
+
+    constructor() {
+        this.emailObjects = {};
     }
 
-    /**
-      *  Called when the signed in status changes, to update the UI
-      *  appropriately. After a sign-in, the API is called.
-      */
-    function updateSigninStatus(isSignedIn) {
-        if (isSignedIn) {
-            authorizeButton.style.display = 'none';
-            signoutButton.style.display = 'block';
-            // getMessage();
-            listMessages();
-        }   else {
-            authorizeButton.style.display = 'block';
-            signoutButton.style.display = 'none';
+
+    getEmailBody(decodedEmail) {
+        const emailBodyStartIndex = decodedEmail.indexOf("Content-Type: text/plain; charset=\"UTF-8\"");
+        const emailBodyEndIndex = (decodedEmail.substring(emailBodyStartIndex)).indexOf("Content-Type: text/html; charset=\"UTF-8\"")
+        let emailBodyValue = decodedEmail.substring(emailBodyStartIndex, emailBodyStartIndex + emailBodyEndIndex);
+        if (emailBodyValue.indexOf("Forwarded message") != -1) {
+            emailBodyValue = emailBodyValue.substring(0, emailBodyValue.indexOf("Forwarded message"));
         }
+        const quotedPrintableIndex = emailBodyValue.indexOf("quoted-printable");
+        if (quotedPrintableIndex != -1) {
+            emailBodyValue = emailBodyValue.substring(quotedPrintableIndex + 16);
+        }
+        return emailBodyValue;
     }
 
-    /**
-      *  Sign in the user upon button click.
-      */
-    function handleAuthClick(event) {
-        gapi.auth2.getAuthInstance().signIn();
+    getEmailDate(decodedEmail) {
+        const emailDateStartIndex = decodedEmail.indexOf("Date:");
+        const emailDateEndIndex = decodedEmail.substring(emailDateStartIndex).indexOf("Message-ID");
+        return decodedEmail.substring(emailDateStartIndex, emailDateStartIndex + emailDateEndIndex);
     }
 
-    /**
-      *  Sign out the user upon button click.
-      */
-    function handleSignoutClick(event) {
-        gapi.auth2.getAuthInstance().signOut();
+    getMessageID(decodedEmail) {
+        const MessageIDStartIndex = decodedEmail.indexOf("Message-ID:") + 12;
+        const MessageIDEndIndex = decodedEmail.substring(MessageIDStartIndex).indexOf("Subject:") + MessageIDStartIndex;
+        let emailDateValue = decodedEmail.substring(MessageIDStartIndex, MessageIDEndIndex);
+        if (emailDateValue.indexOf("X-Notifications") != -1) {
+            emailDateValue = emailDateValue.substring(0, emailDateValue.indexOf("X-Notifications"));
+        }
+        return emailDateValue;
     }
 
     /**
-      * Append a pre element to the body containing the given message
-      * as its text node. Used to display the results of the API call.
+      * Checks if the email is a valid email
       *
-      * @param {string} message Text to be placed in pre element.
+      * Criteria 1: Isn't a Daily Insider message - breaks ML model
```

Comment:
Commented on the other PR too but can you remove this check? I think it's better to limit the number of emails sent to the ML instead of excluding certain emails

Reply:
Resolved offline